### PR TITLE
Use host_type/HostType names in mod_mam_rdbms_arch/mod_mam_muc_rdbms_arch

### PR DESCRIPTION
**Pull request by arcusfelis**
This PR addresses "more pretty".

Proposed changes include:
* Use host_type/HostType names in mod_mam_rdbms_arch/mod_mam_muc_rdbms_arch
* No functional changes (it's just about variable names)
* mod_mam_rdbms_prefs is untouched to make the PR small (ok, it also fixes GDPR message extraction)

### CODE REVIEW

File: src/mam/mod_mam_riak_timed_arch_yz.erl [hunk 1]
```diff
@@ -374,15 +374,17 @@ get_message2(Host, MsgId, Bucket, Key) ->
         _ ->
             []
     end.
--spec get_mam_pm_gdpr_data(ejabberd_gen_mam_archive:mam_pm_gdpr_data(), jid:jid()) ->
+-spec get_mam_pm_gdpr_data(ejabberd_gen_mam_archive:mam_pm_gdpr_data(),
+                           mongooseim:host_type(), jid:jid()) ->
     ejabberd_gen_mam_archive:mam_pm_gdpr_data().
-get_mam_pm_gdpr_data(Acc, OwnerJid) ->
+get_mam_pm_gdpr_data(Acc, HostType, OwnerJid) ->
```
Comment (chrzaszcz) [inline src/mam/mod_mam_riak_timed_arch_yz.erl:380]:
```suggestion
get_mam_pm_gdpr_data(Acc, _HostType, OwnerJid) ->
```

File: src/mam/mod_mam_riak_timed_arch_yz.erl [hunk 2]
```diff
@@ -374,15 +374,17 @@ get_message2(Host, MsgId, Bucket, Key) ->
...
     Messages = get_mam_gdpr_data(OwnerJid, <<"pm">>),
     [{Id, jid:to_binary(Jid), exml:to_binary(Packet)} || #{id := Id, jid := Jid, packet := Packet} <- Messages] ++ Acc.
 
--spec get_mam_muc_gdpr_data(ejabberd_gen_mam_archive:mam_muc_gdpr_data(), jid:jid()) ->
+-spec get_mam_muc_gdpr_data(ejabberd_gen_mam_archive:mam_muc_gdpr_data(),
+                            mongooseim:host_type(), jid:jid()) ->
     ejabberd_gen_mam_archive:mam_muc_gdpr_data().
-get_mam_muc_gdpr_data(Acc, JID) ->
+get_mam_muc_gdpr_data(Acc, _HosType, JID) ->
```
Comment (chrzaszcz) [inline src/mam/mod_mam_riak_timed_arch_yz.erl:387]:
```suggestion
get_mam_muc_gdpr_data(Acc, _HostType, JID) ->
```

File: src/mam/mod_mam_rdbms_async_pool_writer.erl [hunk 3]
```diff
@@ -28,9 +28,7 @@
 -behaviour(gen_mod).
 -behaviour(mongoose_module_metrics).
 
--export([archive_size/4,
-         archive_message/3,
-         lookup_messages/3]).
```
Comment (chrzaszcz) [inline src/mam/mod_mam_rdbms_async_pool_writer.erl:33]: The two removed callbacks were completely unused, right?

**Review by chrzaszcz — COMMENTED** [review]
Looks good, only a few minor comments form me.

File: src/mam/mod_mam_rdbms_async_pool_writer.erl (hunk 3 above)
Comment (arcusfelis) [reply src/mam/mod_mam_rdbms_async_pool_writer.erl:33]: right, they were used long time ago to wait for the writer, but we've removed that feature earlier.
there are no hook registration for it.

File: src/mam/mod_mam_riak_timed_arch_yz.erl (hunk 2 above)
Comment (arcusfelis) [reply src/mam/mod_mam_riak_timed_arch_yz.erl]: applied

**Review by chrzaszcz — APPROVED** [review]
It looks good now!

**Pull request merged**